src/compiler/scala/tools/nsc/typechecker/Unapplies.scala

```diff
@@ -156,10 +156,6 @@ trait Unapplies extends ast.TreeDSL {
     val mods =
       if (applyShouldInheritAccess(inheritedMods))
         (caseMods | (inheritedMods.flags & PRIVATE)).copy(privateWithin = inheritedMods.privateWithin)
-          .tap { mods =>
-            if (currentRun.isScala3 && mods != caseMods)
-              runReporting.warning(cdef.pos, "constructor modifiers are assumed by synthetic `apply` method", WarningCategory.Scala3Migration, cdef.symbol)
-          }
```

Comment:
so this code runs even if there is a user-defined companion `apply`? where do we check for that case?

Reply:
there are many comments in `applyUnapplyMethodCompleter` I haven't read yet.